### ISSUE TITLE
Too much squish, switch offset not working properly 

### ISSUE THREAD

**Issue by Wishingtreetops** [protoloft/klipper_z_calibration#118]
I'm not sure if it's the latest update that broke this, or what. I set my z position end stop, after leveling everything. And I manually enter it into my printer config. Then I do my first layer test in micro step as needed. For my current material my switch offset is usually 0.45, for some reason now I'm having to drop it all the way down to 09, it's as if something within the z Cal is throwing off my numbers. I've asked on forums and asked him discords and has gotten nowhere

**Comment by TitusLabs** [comment]
This should not be caused by any update. But, it's hard to say. May be you have plastic left on the nozzle or your switch is somehow damaged?

**Comment by jinnn1989** [comment]
I am also facing the same issue. Sometimes the first layer is perfect while sometime its too squished but I know for sure this issue is happening after update 1.0

**Comment by TitusLabs** [comment]
Strange, I've had the same problem the last few times. But for me it was the very first time. It has nothing to do with the update. I had blamed it on the fact that I had printed PLA for the first time, which was very ooze. However, the error only seemed to go away after a Klipper restart :thinking: 

The typical Omron switch can develop a drift over time. I have had fewer problems with this so far, but some people have reported it.

**Comment by jinnn1989** [comment]
I did try with Omron and Other switches as well but still I see the issue. Earlier my Z-offset was 0.65 for 0.4 nozzle and 0.63 for 0.8 Nozzle and that would work everytime. but not for 0.4 nozzle even 0.17 does not work. 

Not sure what is causing the issue but everytime I need to look for the squeeze. 

Just some results below with different z offset, first time it comes very well, then it gets squeezed.

Z-CALIBRATION: probe=6.007 - (switch=8.556 - nozzle=2.582 + switch_offset=0.190) --> new offset=-0.156250

Z-CALIBRATION: probe=6.042 - (switch=8.592 - nozzle=2.581 + switch_offset=0.190) --> new offset=-0.158750

Z-CALIBRATION: probe=6.030 - (switch=8.576 - nozzle=2.574 + switch_offset=0.190) --> new offset=-0.162500

Z-CALIBRATION: probe=6.052 - (switch=8.599 - nozzle=2.575 + switch_offset=0.190) --> new offset=-0.161250

Z-CALIBRATION: probe=6.246 - (switch=8.781 - nozzle=2.740 + switch_offset=0.180) --> new offset=0.024750

Z-CALIBRATION: probe=6.245 - (switch=8.772 - nozzle=2.743 + switch_offset=0.180) --> new offset=0.036000

**Comment by jinnn1989** [comment]
more results below, this is after I start the printer, give it a print then I see the squeeze is too much, so cancel the print and restart the same print and again issue .

Z-CALIBRATION: probe=6.242 - (switch=8.748 - nozzle=2.745 + switch_offset=0.180) --> new offset=0.058500

Z-CALIBRATION: probe=6.197 - (switch=8.738 - nozzle=2.747 + switch_offset=0.180) --> new offset=0.026000

**Comment by TitusLabs** [comment]
OK, the result is still consistent for me. There were really only a few exceptions. If you have constantly changing values, then there is something mechanically wrong with your printer. E.g. crooked/wobbling magnets, loose pulleys, wobbling print head, oozing nozzle.... just to name a few. Is your PROBE_ACCURACY consistent?

**Comment by jinnn1989** [comment]
Yes prob accuracy is was consistent, very slight variation which I expect it. but rest it was good. I need to check other mechanical parts. but for sure they are not an issues.

**Comment by jinnn1989** [comment]
Z-CALIBRATION: probe=5.332 - (switch=7.511 - nozzle=1.444 + switch_offset=0.250) --> new offset=-0.984272

Z-CALIBRATION: probe=5.334 - (switch=7.511 - nozzle=1.447 + switch_offset=0.250) --> new offset=-0.979272

Z-CALIBRATION: probe=5.333 - (switch=7.509 - nozzle=1.449 + switch_offset=0.250) --> new offset=-0.976772

these were my prob calibration reults

**Comment by TitusLabs** [comment]
These values look pretty good. Now tune your switch_offset to barely touch the bed with the nozzle (the real z=0 position and no paper or something). Then test it again. And maybe even restart Klipper and test it again. Be careful with meshes if you use it. Start without a mesh and recalibrate it after you have calibrated your z...

**Comment by jinnn1989** [comment]
Yes the calibration values are good, I did a slight change in my Start gcode, that was to move the z calibration before mesh, and I think i am starting to get good results. but still the values are not that stable. Tomorrow will check mechanical parts.

once more thing I observed was that on PEI smooth bed the results are very good, but on texture its not that stable.

**Comment by TitusLabs** [comment]
Ok, on textured PEI the consistency depends on the switch and how fine the texture is. Unfortunately, there aren't that many plates that really have a fine texture.

**Comment by buhralex** [comment]
I basically have the same problem as @jinnn1989. I dialed in the offset many times. Sometimes the print turns out really good, next print I get an elephant foot or the filament doesn't stick to the buildplate at all.

I already checked the mechanical parts and found no issues.
When I deactivate Auto Z and dial in my Offset manually I get perfect and consistent results.

I've read somewhere that this could have something to do with me using KAMP but even when I restart Klipper, clear the mesh, home the printer and then do a calibrate_z the results vary and my nozzle doesn't touch the buildplate or dives right into it.

**Comment by TitusLabs** [comment]
@buhralex is this still an issue? Do you have this problem with a textured plate or even with a smooth PEI?

**Comment by Wishingtreetops** [comment]
> I basically have the same problem as @jinnn1989. I dialed in the offset many times. Sometimes the print turns out really good, next print I get an elephant foot or the filament doesn't stick to the buildplate at all.
> 
> I already checked the mechanical parts and found no issues. When I deactivate Auto Z and dial in my Offset manually I get perfect and consistent results.
> 
> I've read somewhere that this could have something to do with me using KAMP but even when I restart Klipper, clear the mesh, home the printer and then do a calibrate_z the results vary and my nozzle doesn't touch the buildplate or dives right into it.

Did you ever have it working previously? ... Have you tried removing/disabling KAMP to see if that helps you further progress? 

What rendition of klicky do you use? ... Long shot possibly, and maybe you're not at that point yet,  but once I had gotten klicky and my Auto z to work smoothly, after some time the numbers would slowly start drifting off again, so i also had to replace the plastic housings I had printed for the klicky base . . . That's when I decided to ultimately switch to PCB klicky.. And it was the best upgrade I could have ever made, I have had zero Auto z or leveling issues since then, and that was A little over 6 months ago.

**Comment by TitusLabs** [comment]
Closing this long-running thread, with the reasoning written down here:
https://github.com/protoloft/klipper_z_calibration/wiki/Troubleshooting#the-squish-changes-between-prints-or-between-build-plates

`CALIBRATE_Z` measures the print surface on every run, so a plate swap is
compensated — thickness and surface alike. In practice the result stays
consistent across build plates, coarse textured ones included. A first layer that
changes from print to print is a mechanical inconsistency, and worn or
poor-quality switches are high on that list: a microswitch's trigger point drifts
as it ages, on the calibration endstop as much as on the probe. Play in the probe
mount or dock, a sticking endstop pin, and residue on the nozzle do the same.

@jinnn1989's case is the typical shape of it — consistent samples within a run,
different results the next day. A clean `PROBE_Z_ACCURACY` doesn't rule that out;
compare several runs taken across sessions instead.

If you just want a different squish on a given surface, apply it on top of the
calibrated offset with `SET_GCODE_OFFSET Z_ADJUST=... MOVE=1` rather than
changing `switch_offset`:
https://github.com/protoloft/klipper_z_calibration/wiki/How-To-Use-It#optional-slicer-adjustment
